var/spack/repos/builtin/packages/ngspice/package.py

```diff
@@ -10,14 +11,121 @@ class Ngspice(AutotoolsPackage):
 
     homepage = "http://ngspice.sourceforge.net/"
     url      = "https://sourceforge.net/projects/ngspice/files/ngspice-33.tar.gz"
+    git      = "git://git.code.sf.net/p/ngspice/ngspice"
 
+    maintainers = ['aweits', 'cessenat']
+
+    # Master version by default adds the experimental adms feature
+    version('master', branch='master')
     version('34', sha256='2263fffc6694754972af7072ef01cfe62ac790800dad651bc290bfcae79bd7b5')
     version('33', sha256='b99db66cc1c57c44e9af1ef6ccb1dcbc8ae1df3e35acf570af578f606f8541f1')
+    version('32', sha256='3cd90c4e94516d87c5b4d02a3a6405b1136b25d05c871d4fee1fd7c4c0d03ef2')
+    version('31', sha256='845f3b0c962e47ded051dfbc134c3c1e4ac925c9f0ce1cb3df64eb9b9da5c282')
+    version('30', sha256='08fe0e2f3768059411328a33e736df441d7e6e7304f8dad0ed5f28e15d936097')
+    version('29', sha256='8d6d0ffbc15f248eb6ec3bde3b9d1397fbc95cb677e1c6a14ff46065c7f95c4a')
+    version('27', sha256='0c08c7d57a2e21cf164496f3237f66f139e0c78e38345fbe295217afaf150695')
+
+    # kicad needs build=lib, i.e. --with--ngshared
+    variant(
+        'build', default='lib',
+        description='Build type: lib=ngshared, bin otherwise',
+        values=('lib', 'bin'),
+        multi=False,
+    )
+
+    variant('X', default=False, description='Use the X Window System')
+    variant(
+        'debug', default='auto',
+        description='Enable debugging features: '
+        'auto is yes for build=lib, no for build=bin',
+        values=('auto', 'yes', 'no'),
+        multi=False,
+    )
+    variant('xspice', default=False, description='Enable XSPICE enhancements')
+    variant('cider', default=False, description='Enable CIDER enhancements')
+    variant('openmp', default=False, description='Compile with multi-threading support')
+    variant('readline', default=True, description='Build readline support (for bin)')
+    variant('fft', default=True, description='Use external fftw lib')
+
+    depends_on('fftw-api@3:~mpi~openmp', when='+fft~openmp')
+    depends_on('fftw-api@3:~mpi+openmp', when='+fft+openmp')
+    depends_on('readline', when='+readline build=bin')
+
+    # Needed for autoreconf:
+    depends_on('bison', type='build', when='@master')
+    depends_on('flex', type='build', when='@master')
+
+    # INSTALL indicates dependency on these but they are underlying
+    # depends_on('freetype', when='+X build=bin')
+    # depends_on('libxrender', when='+X build=bin')
+    # depends_on('fontconfig', when='+X build=bin')
+    # depends_on('libxft', when='+X build=bin')
+    # depends_on('libxext', when='+X build=bin')
+    # depends_on('libxmu', when='+X build=bin')
+    depends_on('libxaw', when='+X build=bin')
```

Comment:
I have to enable at least freetype/fontconfig/libxft to get an install that doesn't pull in the system versions of these.